akka-typed/src/main/scala/akka/typed/scaladsl/Actor.scala

```diff
@@ -286,41 +286,45 @@ object Actor {
    * system to reuse the previous behavior. This is provided in order to
    * avoid the allocation overhead of recreating the current behavior where
    * that is not necessary.
+   *
+   * Same as [[Behavior.Same]], but replicated here for convenienve when using the the Actor DSL via import `Actor._`.
    */
-  def Same[T]: Behavior[T] = SameBehavior.asInstanceOf[Behavior[T]]
+  def Same[T]: Behavior[T] = Behavior.Same
 
   /**
    * Return this behavior from message processing in order to advise the
    * system to reuse the previous behavior, including the hint that the
    * message has not been handled. This hint may be used by composite
    * behaviors that delegate (partial) handling to other behaviors.
+   *
+   * Same as [[Behavior.Unhandled]], but replicated here for convenienve when using the the Actor DSL via import `Actor._`.
    */
-  def Unhandled[T]: Behavior[T] = UnhandledBehavior.asInstanceOf[Behavior[T]]
-
-  /*
-   * TODO write a Behavior that waits for all child actors to stop and then
-   * runs some cleanup before stopping. The factory for this behavior should
-   * stop and watch all children to get the process started.
-   */
+  def Unhandled[T]: Behavior[T] = Behavior.Unhandled
 
   /**
    * Return this behavior from message processing to signal that this actor
    * shall terminate voluntarily. If this actor has created child actors then
    * these will be stopped as part of the shutdown procedure. The PostStop
    * signal that results from stopping this actor will NOT be passed to the
    * current behavior, it will be effectively ignored.
+   *
+   * Same as [[Behavior.Stopped]], but replicated here for convenienve when using the the Actor DSL via import `Actor._`.
    */
-  def Stopped[T]: Behavior[T] = StoppedBehavior.asInstanceOf[Behavior[T]]
+  def Stopped[T]: Behavior[T] = Behavior.Stopped
 
   /**
    * A behavior that treats every incoming message as unhandled.
+   *
+   * Same as [[Behavior.Empty]], but replicated here for convenienve when using the the Actor DSL via import `Actor._`.
    */
-  def Empty[T]: Behavior[T] = EmptyBehavior.asInstanceOf[Behavior[T]]
+  def Empty[T]: Behavior[T] = Behavior.Empty
 
   /**
    * A behavior that ignores every incoming message and returns “same”.
+   *
+   * Same as [[Behavior.Ignore]], but replicated here for convenienve when using the the Actor DSL via import `Actor._`.
    */
-  def Ignore[T]: Behavior[T] = IgnoreBehavior.asInstanceOf[Behavior[T]]
+  def Ignore[T]: Behavior[T] = Behavior.Ignore
 
   /**
```

Comment:
the type `Behavior` and `ExtensibleBehavior` are independent of language, while `scaladsl.Actor` and `javadsl.Actor` contains the language specifics. If something is not core it could be moved out of `Actor` but it would still have to live in `scaladsl` and `javadsl`.